cpp/src/arrow/util/meson.build

```diff
@@ -22,7 +22,6 @@ conf_data.set('ARROW_VERSION_MINOR', version_minor)
 conf_data.set('ARROW_VERSION_PATCH', version_patch)
 
 conf_data.set('ARROW_VERSION', arrow_version)
-conf_data.set('ARROW_VERSION_STRING', arrow_version)
```

Comment:
Why do we need to remove this?

Reply:
This configuration data is unused. In the cmake file, there is a `#define ARROW_VERSION_STRING @ARROW_VERSION@` but nothing for `@ARROW_VERSION_STRING@`

Reply:
Ah, I see!